src/subtyping.md

```diff
@@ -1,146 +1,127 @@
+        debug(hello, world);
+    }
 }
 ```
 
-By default, static types must match *exactly* for a program to compile. As such,
-this code won't compile:
+In an overly restrictive implementation of lifetimes, since `a` and `b` have differeing lifetimes,
+we might see the following error:
 
-<!-- ignore: simplified code -->
-```rust,ignore
-let mr_snuggles: Cat = ...;
-love(mr_snuggles);         // ERROR: expected Animal, found Cat
+```text
+error[E0308]: mismatched types
+ --> src/main.rs:10:16
+   |
+10 |         debug(hello, world);
+   |                      ^
+   |                      |
+   |                      expected `&'static str`, found struct `&'b str`
 ```
 
-Mr. Snuggles is a Cat, and Cats aren't *exactly* Animals, so we can't love him! 😿
+This is over-restrictive. In this case, what we want is to accept any type that lives *at least as long* as `'b`.
+Let's try using subtyping with our lifetimes.
 
-This is annoying because Cats *are* Animals. They support every operation
-an Animal supports, so intuitively `love` shouldn't care if we pass it a `Cat`.
-We should be able to just **forget** the non-animal parts of our `Cat`, as they
-aren't necessary to love it.
+Let's define a lifetime to have the a simple set of requirements: `'a` defines a region of code in which a value will be alive.
+Now that we have a defined set of requirements for lifetimes, we can define how they relate to each other.
+`'a: 'b` if and only if `'a` defines a region of code that **completely contains** `'b`.
 
-This is exactly the problem that *subtyping* is intended to fix. Because Cats are just
-Animals **and more**, we say Cat is a *subtype* of Animal (because Cats are a *subset*
-of all the Animals). Equivalently, we say that Animal is a *supertype* of Cat.
-With subtypes, we can tweak our overly strict static type system
-with a simple rule: anywhere a value of type `T` is expected, we will also
-accept values that are subtypes of `T`.
+`'a` may define a region larger than `'b`, but that still fits our definition.
+Going back to our example above, we can say that `'static: 'b`.
 
-Or more concretely: anywhere an Animal is expected, a Cat or Dog will also work.
+For now, let's accept the idea that subtypes of lifetimes can be passed through references (more on this in [Variance](#variance)),
+eg. `&'static str` is a subtype of `&'b str`, then we can let them coerce, and then the example above will compile
 
-As we will see throughout the rest of this section, subtyping is a lot more complicated
-and subtle than this, but this simple rule is a very good 99% intuition. And unless you
-write unsafe code, the compiler will automatically handle all the corner cases for you.
+```rust
+fn debug<T: std::fmt::Debug>(a: T, b: T) {
+    println!("a = {:?} b = {:?}", a, b);
+}
 
-But this is the Rustonomicon. We're writing unsafe code, so we need to understand how
-this stuff really works, and how we can mess it up.
```

Comment:
I actually really liked this section, can we keep it? I think it's important to say this stuff is important when writing unsafe. And maybe include an example of messing it up?

Reply:
Absolutely. It must have slipped past me while I was cleaning it up